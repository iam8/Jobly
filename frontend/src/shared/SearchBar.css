.SearchBar .col {
    padding-right: 0px;
}