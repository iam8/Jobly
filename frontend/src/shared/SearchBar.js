import React, {useState} from "react";
import {Button, Form, Label, Input, Row, Col} from "reactstrap";

import "./SearchBar.css";


/**
 * Search bar. Used on /companies and /jobs pages to allow filtering.
 *
 * Calls the searchFor method (passed by parent), which performs the actual filtering.
 */
function SearchBar({searchFor}) {
    const [searchTerm, setSearchTerm] = useState("");

    /** Call searchFor() to perform the filtering. */
    const handleSubmit = (evt) => {
        evt.preventDefault();
        searchFor(searchTerm.trim() || undefined);
        setSearchTerm(searchTerm.trim());
    }

    /** Update search bar field. */
    const handleChange = (evt) => {
        setSearchTerm(evt.target.value);
    }

    return (
        <div className="SearchBar mb-4">
            <Form onSubmit={handleSubmit}>
                <Row>
                    <Col>
                        <Label
                            className="visually-hidden"
                            for="search-input"
                        >
                            Search
                        </Label>

                        <Input
                            id="search-input"
                            name="searchTerm"
                            placeholder="Enter a search term"
                            value={searchTerm}
                            onChange={handleChange}
                            bsSize="lg"
                        />
                    </Col>

                    <Col>
                        <Button type="submit" color="primary" size="lg">Search</Button>
                    </Col>
                </Row>
            </Form>
        </div>
    )
}


export default SearchBar;
