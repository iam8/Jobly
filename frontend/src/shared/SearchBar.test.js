import React from "react";
import { render } from "@testing-library/react";

import SearchBar from "./SearchBar";


test("Renders without crashing", () => {
    render(<SearchBar />);
})


test("Matches snapshot", () => {
    const { asFragment } = render(<SearchBar />);
    expect(asFragment()).toMatchSnapshot();
})
