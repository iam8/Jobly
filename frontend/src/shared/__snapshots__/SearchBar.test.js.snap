// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="SearchBar mb-4"
  >
    <form
      class=""
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <label
            class="visually-hidden form-label"
            for="search-input"
          >
            Search
          </label>
          <input
            class="form-control-lg form-control"
            id="search-input"
            name="searchTerm"
            placeholder="Enter a search term"
            type="text"
            value=""
          />
        </div>
        <div
          class="col"
        >
          <button
            class="btn btn-primary btn-lg"
            type="submit"
          >
            Search
          </button>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
